lib/rules/no-children-prop.js

```diff
@@ -36,15 +36,38 @@ module.exports = {
       recommended: true,
       url: docsUrl('no-children-prop')
     },
-    schema: []
+    schema: [{
+      type: 'object',
+      properties: {
+        allowFunctions: {
+          type: 'boolean',
+          default: false
+        }
+      },
+      additionalProperties: false
+    }]
   },
   create: function(context) {
+    const configuration = context.options[0] || {};
+
+    function isFunction(node) {
+      return (node.type === 'ArrowFunctionExpression' || node.type === 'FunctionExpression')
```

Comment:
what about generators or async functions? is there a more generic way to catch all functions?

Reply:
According to AST explorer, both of these are `FunctionExpression`s with either `generator` or `async` flag equal to true.

Reply:
awesome, thanks. maybe test cases with these would be good tho :-D